src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java

```diff
@@ -63,44 +63,47 @@ public final class DTLogWriter implements Closeable, Flushable {
   private final FileChannel channel;
   private final ByteBuffer  buffer;
   private final BitSet      bitSet;
+  private final DTLog.Level level;
 
   private long lastTimestamp;
   private int  nextVarHandle = 0x0100;
 
-  private DTLogWriter() throws IOException {
+  private DTLogWriter(DTLog.Level logLevel) throws IOException {
     Instant now = Clock.systemUTC()
                        .instant();
-    lastTimestamp = RobotController.getFPGATime() / 1_000;
+    lastTimestamp = DTRobot.currentTimeMicros() / 1000;
     long startTimeMillis = now.toEpochMilli();
     String name = "LOG_" + TIME_FORMATTER.format(now) + ".dtlog";
 
     file = new File(LOG_DIRECTORY, name);
     channel = FileChannel.open(file.toPath(), WRITE, CREATE, TRUNCATE_EXISTING);
-    buffer = ByteBuffer.allocateDirect(BUFFER_SIZE);
+    buffer = ByteBuffer.allocateDirect(BUFFER_SIZE_BYTES);
     bitSet = new BitSet();
+    level = logLevel;
 
     writeBytes(HEADER_MAGIC_BYTES);
     int checksum = HEADER_MAGIC_XOR;
 
-    int dtlibVersion = (DTLibInfo.Version.YEAR << 16) | (DTLibInfo.Version.MAJOR << 8)
-        | DTLibInfo.Version.MINOR;
+    int dtlibVersion =
+        (DTLibInfo.Version.YEAR << 16) | (DTLibInfo.Version.MAJOR << 8) | DTLibInfo.Version.MINOR;
     writeInt(dtlibVersion);
     checksum ^= dtlibVersion;
 
-    String[] wpilibVersions = WPILibVersion.Version.split("\\.");
+    String[] wpilibVersions = WPILibVersion.Version.substring(0, WPILibVersion.Version.indexOf('-'))
+                                                   .split("\\.");
     int wpilibYear = Integer.parseInt(wpilibVersions[0]);
     int wpilibMajor = Integer.parseInt(wpilibVersions[1]);
     int wpilibMinor = Integer.parseInt(wpilibVersions[2]);
     int wpilibVersion = (wpilibYear << 16) | (wpilibMajor << 8) | wpilibMinor;
     writeInt(wpilibVersion);
     checksum ^= wpilibVersion;
 
-    int team = getTeamNumber();
-    writeLong(((long) team << 48) | startTimeMillis);
+    long team = DTRobot.getTeamNumber();
+    writeLong((team << 48) | startTimeMillis);
     checksum ^= team << 16;
     checksum ^= (int) ((startTimeMillis >> 32) | startTimeMillis);
-
     writeInt(checksum);
+
```

Comment:
You might consider something like CRC32 for the checksum: https://docs.oracle.com/javase/8/docs/api/java/util/zip/CRC32.html